### DISCUSSION TITLE
How to execute native apps？[bug]

### DISCUSSION THREAD

**Discussion by yeyu12** [tauri-apps/tauri#5609] [Q&A]
### Describe the bug

Use tauri as the control panel. There are multiple applications in the panel. The user downloads the application from the panel and then opens it through the panel. The business requirements are like this. I tried to configure the relevant instructions according to the shell in the documentation, but all failed to open (js). Please, how do I configure it, or how do I achieve it. I would not use rust

### Platform and versions

Environment
  › OS: Windows 10.0.22000 X64
  › Webview2: 107.0.1418.35
  › MSVC: 
      - Visual Studio Community 2022
  › Node.js: 16.13.2
  › npm: 8.9.0
  › pnpm: Not installed!
  › yarn: 1.22.17
  › rustup: 1.25.1
  › rustc: 1.63.0
  › cargo: 1.63.0
  › Rust toolchain: stable-x86_64-pc-windows-msvc 

Packages
  › @tauri-apps/cli [NPM]: 1.1.1
  › @tauri-apps/api [NPM]: 1.1.0
  › tauri [RUST]: 1.1.1,
  › tauri-build [RUST]: 1.1.1,
  › tao [RUST]: 0.14.0,
  › wry [RUST]: 0.21.1,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../dist
  › devPath: http://localhost:8000/
  › framework: React

App directory structure
  ├─ .git
  ├─ .husky
  ├─ mock
  ├─ node_modules
  ├─ src
  ├─ src-tauri
  └─ swagger

**Comment by Blatman** [comment]
Not sure on Windows if it is similar but this is how I do it on Mac - basically I have a bunch of app icons in a panel (launcher) and click on an icon to run the app.
eg: app called hippo -

tauri.config.json:
  build parameters -  "withGlobalTauri": true

"shell": {
        "execute": true,
        "scope": [
		  {
                   "name": "startapp",
                   "cmd": "open",
		   "sidecar": false,
                   "args": true
          },

JS:
    var cmd = window.__TAURI__.shell.Command  (need to set in config to use this method)

// scope needs to be set for child processes
    var startApp = new cmd('startapp', ['-a', 'hippo'], {cwd: './'}).execute();  // need to use open to allow Apple apps to work

  equiv. to   'open -a hippo' in terminal

I find most hassle with child process stuff in Tauri is the scoping requirements - so maybe a clue there.

**Reply by yeyu12** [reply]
Thank you very much. I have used other solutions here to solve this problem. 
Below is my code
[config](tauri.config.json:)
 "shell": {
                "all": true,
                "execute": true,
                "open": true,
                "sidecar": true,
                "scope": [{
                    "name": "windows",
                    "cmd": "cmd.exe",
                    "args": [
                        "/C",
                        {
                            "validator": "\\S+"
                        }
                    ]
                }]
            }
js:
const commandServer = new Command('windows', [
    '/C',
    'application instruction to execute',
  ]);
  commandServer.stdout.on('data', line => console.log(`response: "${line}"`));
  commandServer.stderr.on('data', line => console.log(`command stderr: "${line}"`));
  commandServer.spawn();
